src/micromamba/install.sh

```diff
@@ -1,101 +1,42 @@
 #!/usr/bin/env bash
 
-VERSION=${VERSION:-"latest"}
-
-USERNAME=${USERNAME:-"automatic"}
-
 set -e
+FEATURE_DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
+cd "${FEATURE_DIR}"
 
-# Clean up
-rm -rf /var/lib/apt/lists/*
-
-if [ "$(id -u)" -ne 0 ]; then
-    echo -e 'Script must be run as root. Use sudo, su, or add "USER root" to your Dockerfile before running this script.'
-    exit 1
-fi
+# Options
+VERSION=${VERSION:-"latest"}
+REINSTALL=${REINSTALL:-"false"}
+ADD_CONDA_FORGE=$ADDCONDAFORGE
 
-architecture="$(dpkg --print-architecture)"
-if [ "${architecture}" != "amd64" ] && [ "${architecture}" != "arm64" ]; then
-    echo "(!) Architecture $architecture unsupported"
-    exit 1
-fi
+# Constants
+MAMBA_ROOT_PREFIX="/opt/conda"
+micromamba_destination="/usr/local/bin"
 
-# Determine the appropriate non-root user
-if [ "${USERNAME}" = "auto" ] || [ "${USERNAME}" = "automatic" ]; then
-    USERNAME=""
-    POSSIBLE_USERS=("vscode" "node" "codespace" "$(awk -v val=1000 -F ":" '$3==val{print $1}' /etc/passwd)")
-    for CURRENT_USER in "${POSSIBLE_USERS[@]}"; do
-        if id -u "${CURRENT_USER}" >/dev/null 2>&1; then
-            USERNAME=${CURRENT_USER}
-            break
-        fi
-    done
-    if [ "${USERNAME}" = "" ]; then
-        USERNAME=root
-    fi
-elif [ "${USERNAME}" = "none" ] || ! id -u ${USERNAME} >/dev/null 2>&1; then
-    USERNAME=root
-fi
+# shellcheck source=./utils.sh
+source ./utils.sh
 
-apt_get_update() {
-    if [ "$(find /var/lib/apt/lists/* | wc -l)" = "0" ]; then
-        echo "Running apt-get update..."
-        apt-get update -y
-    fi
-}
+USERNAME="${USERNAME:-"${_REMOTE_USER:-"automatic"}"}"
+detect_user USERNAME
 
-# Checks if packages are installed and installs them if not
-check_packages() {
-    if ! dpkg -s "$@" >/dev/null 2>&1; then
-        apt_get_update
-        apt-get -y install --no-install-recommends "$@"
-    fi
-}
+require_running_as_root
 
-check_git() {
-    if [ ! -x "$(command -v git)" ]; then
-        check_packages git
-    fi
+ensure_download_prerequisites() {
+    # This is the only place we need to use apt, so we can scope clean_up_apt tightly:
+    clean_up_apt
+    check_packages curl ca-certificates bzip2
+    clean_up_apt
```

Comment:
For understandability, I think this process should be executed at the end of the script.

Reply:
For the initial cleanup, I added a comment to explain that it's performed as necessary.

The state of the cache is tracked with the `apt_cache_state` variable which can have values `unaccessed`, `updated`, or `clean`.

For the final cleanup, I renamed it to `clean_up_apt_if_updated`.

Is this good?

Reply:
To me, that seems to overcomplicate things.
It may be a matter of preference, so if you want to go with this, that's fine.

Reply:
Yes, I feel conflicted here. On one hand I don't want unnecessary complication. On the other hand, I am annoyed that `install.sh` scripts delete the apt cache even when not necessary. For example, what if I frequently build base containers and I want to prepopulate the apt-cache? It seems like a strange side-effect.

Reply:
I believe the Dockerfile best practices state that this type of cache needs to be removed on a layer-by-layer basis.
So I see no need to take into concern the cache that was originally left behind (not following best practices).

Reply:
I wouldn't expect people making custom devcontainers for private use to be following best practices. I added more comments to explain in detail how everything works. Unless you feel strongly I'd prefer to keep it.